xwiki-commons-core/xwiki-commons-job/xwiki-commons-job-default/src/main/java/org/xwiki/job/internal/DefaultJobStatusStore.java

```diff
@@ -211,13 +209,11 @@ private String encode(String name)
         String encoded;
 
         if (name != null) {
-            try {
-                encoded = URLEncoder.encode(name, DEFAULT_ENCODING);
-            } catch (UnsupportedEncodingException e) {
-                // Should never happen
-
-                encoded = name;
-            }
+            encoded = URLEncoder.encode(name, StandardCharsets.UTF_8)
+                // Replace characters that might be problematic in file systems.
+                .replace("+", "%20")
+                .replace(".", "%2E")
+                .replace("*", "%2A");
```

Comment:
`FileSystemStoreUtils#encode` is a bit more complete and focuses on this kind of stuff (while still being based on URL encoding), but it encodes more (to avoid collisions in case-insensitive filesystems it encodes uppercase characters, for example).

Reply:
Indeed, it probably makes sense to use this here, too, so we have one central place to fix if there should be any problem.

Reply:
I just noticed that `FileSystemStoreUtils` is in `xwiki-platform`… Any suggestions where to move it in `xwiki-commons`?

Reply:
Hmm, I guess we could either move it to xwiki-commons-text, or start a xwiki-commons-store module (we will probably have more stuff in there, thinking for example about the plan to provide an API to ask as a proxy to store files either in the filesystem or an object storage).

Reply:
I've moved it to `xwiki-commons-text` for now as it feels strange to create a new module for an internal class. We can move it later when implementing that API.